tests/turnstile_chain.c

```diff
@@ -0,0 +1,106 @@
+#include <ktest.h>
+#include <klog.h>
+#include <mutex.h>
+#include <runq.h>
+#include <sched.h>
+#include <thread.h>
+
+// length of the propagation chain
+#define T 4
+
+// TODO replace asserts about priority borrowing with some accessor
+/* TODO consider where we could check restoring priorities
+ * I think that all threads should keep HIGH priority until the last propagator
+ * has acquired 2nd mutex and then the priorities would be restored to initial
+ * ones (propagator_priority(i) for propagator[i]).
+ */
+
+static mtx_t mtx[T + 1];
+// TODO consider whether this would be any useful
+// volatile int mtx_passed[T + 1];
```

Comment:
I think that it wouldn't because fields in `mtx_t` are already volatile.

Reply:
Sorry, I've misread your comment. Could you explain `mtx_passed`?

Reply:
I meant it as "whether propagator[n] has entered (and possibly already exit) mtx[n]". Pretty much `mtx_acquired` in the deleted test.